### PR TITLE
feat!: create method for GRCh38 to MANE protein representation

### PR DESCRIPTION
**Pull request by korikuzma**
Addresses #151 for GRCh38 (will do liftover/check at a later time). This is needed for VarCat

Notes:
* `MANETranscript` updates:
  * Add method `grch38_to_mane_p`
    * Given GRCh38 representation, will return MANE protein representation. Also allows for longest compatible remaining representation to be returned if no compatible MANE representation found.  
  *  Updates `get_longest_compatible_transcript` 
      * `gene` is now optional
      * Adds `end_annotation_layer` as a parameter. This will allow you to either return protein or cdna representation when set. 
* `MANETranscriptMappings` updates: 
  * `get_gene_mane_data` always returns list. If no MANE data found, will return empty list (previously returned None)
* `UTADatabase` updates:
  * `get_transcripts_from_gene` renamed to `get_transcripts`
    * `gene` is now an optional parameter

### CODE REVIEW

**Comment by korikuzma** [comment]
Added test for #212 which pointed out a bug for braf v600e

**Comment by korikuzma** [comment]
This refactor work has made me think about #63, specifically [this](https://github.com/GenomicMedLab/cool-seq-tool/issues/63#issuecomment-1765387719) comment

**Comment by korikuzma** [comment]
Done making changes. Ready for review!

File: cool_seq_tool/mappers/mane_transcript.py [hunk 1]
```diff
@@ -1005,3 +1022,93 @@ async def g_to_mane_c(
...
+        start_pos: int,
+        end_pos: int,
+        gene: Optional[str] = None,
+        residue_mode: ResidueMode = ResidueMode.RESIDUE,
+        try_longest_compatible: bool = False,
+    ) -> Optional[Dict]:
+        """Given GRCh38 genomic representation, return protein representation.
+        Will try MANE Select and then MANE Plus Clinical. If neither is found and
+        `try_longest_compatible` is set to `true`, will also try to find the longest
+        compatible remaining representation.
```
Comment (jsstevenson) [inline cool_seq_tool/mappers/mane_transcript.py:1038]: Fwiw, sometimes Sphinx API page generation bugs out if you don't include an extra blank line between the descriptions and the params, so I've been trying to include them when I remember (for public methods)

**Review by jsstevenson — APPROVED** [review]
lgtm, two small things

1) I think changing arguments for public methods probably entails an API break (i.e. a "!")
2) made a note about including an extra linebreak in docstrings for public methods/any methods that could conceivably have API pages generated for them

File: cool_seq_tool/mappers/mane_transcript.py (hunk 1 above)
Comment (korikuzma) [reply cool_seq_tool/mappers/mane_transcript.py:1038]: Cool. I usually add a new line, but must've forgot for this one

**Comment by korikuzma** [comment]
Need to actually return both cdna and protein representation

**Comment by korikuzma** [comment]
Actually, I'm going to hold off on this

**Review by jsstevenson — APPROVED** [review]

**Pull request merged**